src/Symfony/Component/Console/Input/InputInterface.php

```diff
@@ -18,6 +18,10 @@
  * InputInterface is the interface implemented by all input classes.
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @method getRawArguments(bool $strip = false): array<string|bool|int|float|null|array<string|bool|int|float|null>> Returns all the given arguments NOT merged with the default values.
+ * @method getRawOptions(): array<string|bool|int|float|array<string|bool|int|float|null>|null> Returns all the given options NOT merged with the default values.
+ * @method unparse(): list<string> Returns a stringified representation of the options passed to the command.
  */
 interface InputInterface
```

Comment:
To avoid any BC break, we can add a new interface for these new capabilities:

```php
interface RawInputInterface extends InputInterface
{
    public function getRawArguments(): array:
    public function getRawOptions(): array;
}
```

Reply:
Would that really help? In the end what you receive, because that is what is typed everywhere, is `InputInterface`. Having a separate interface would bring back something similar to this code:

```php
    public static function getRawOptions(InputInterface $input): array
    {
        return $input instanceof RawInputInterface
            ? $input->getRawOptions()
            : [];
    }
```

Which... err yes in practice if we make `Input` implement it it will work 99% of the cases but type-wise leaves it to be desired.

Reply:
Just to clarify: the only affected users are users implementing `InputInterface` without extending `Input`, which AFAIK (but I can be wrong) is _extremely_ rare, unless you're aware of more cases/usages in the wild?